### PR TITLE
Add REI support for Title Scrolls integration

### PR DESCRIPTION
**Pull request by emilyploszaj**
* Adds an REI plugin that adds queerbric title scrolls as separate entries
* Adds recipes for queerbric title scrolls in REI

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -1,5 +1,5 @@
 plugins {
-	id 'fabric-loom' version '0.4-SNAPSHOT'
+	id 'fabric-loom' version '0.5-SNAPSHOT'
```
Comment (Sturmlilie) [inline build.gradle:2]: Is this just a general upgrade, or essential for REI support?

File: src/main/java/io/github/queerbric/queerbrics/QueerbricsREIPlugin.java [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+import net.minecraft.recipe.Recipe;
+import net.minecraft.util.Identifier;
+import net.minecraft.util.registry.Registry;
+
+@Environment(EnvType.CLIENT)
+public class QueerbricsREIPlugin implements REIPluginV0 {
+	private static List<Pair<QueerBrickItem, EntryStack>> entries = new ArrayList<Pair<QueerBrickItem, EntryStack>>();
+
+	static {
+		if (FabricLoader.getInstance().isModLoaded("titlescrolls")) {
```
Comment (Sturmlilie) [inline src/main/java/io/github/queerbric/queerbrics/QueerbricsREIPlugin.java]: This same check is repeated a couple of times in the code; it probably works but one typo in the title scrolls mod id could be hard to debug, so I usually advise to DRY stuff like this (least intrusive way would just be a static string var somewhere).

File: src/main/java/io/github/queerbric/queerbrics/QueerbricsREIPlugin.java [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+			for (Pair<QueerBrickItem, EntryStack> pair : entries) {
+				recipeHelper.registerDisplay(new ScrollRecipeDisplay(pair));
+			}
+		}
+	}
+
+	class ScrollRecipeDisplay implements DefaultCraftingDisplay {
+		public List<List<EntryStack>> input = new ArrayList<List<EntryStack>>();
+		public List<List<EntryStack>> result = new ArrayList<List<EntryStack>>();
```
Comment (Sturmlilie) [inline src/main/java/io/github/queerbric/queerbrics/QueerbricsREIPlugin.java:79]: I'm not familiar with REI; is this just a simple "put this stuff in, and you get this stuff out" type display?

**Review by Sturmlilie — COMMENTED** [review]
Can't comment on the REI or titlescrolls-specific parts since I'm not familiar with those codebases, but looks okay.

File: src/main/java/io/github/queerbric/queerbrics/QueerbricsREIPlugin.java (hunk 3 above)
Comment (emilyploszaj) [reply src/main/java/io/github/queerbric/queerbrics/QueerbricsREIPlugin.java:79]: In a generic sense, yes, this is a crafting entry you'll see when you check for recipes, these lists are just stored as to not reallocate on every call

File: build.gradle (hunk 1 above)
Comment (emilyploszaj) [reply build.gradle:2]: Both, 0.5 for the types of dependencies when I was initially having problems and at this point no need to downgrade

File: src/main/java/io/github/queerbric/queerbrics/QueerbricsREIPlugin.java (hunk 2 above)
Comment (emilyploszaj) [reply src/main/java/io/github/queerbric/queerbrics/QueerbricsREIPlugin.java]: I changed this to a static string for the checks and identifiers that use it.

**Review by Sturmlilie — APPROVED** [review]
Thanks for also using it for `Identifier` :+1:

**Pull request merged**